pkg/webhook/mutation/pod/metadata/mutator.go

```diff
@@ -141,6 +143,48 @@ func setWorkloadAnnotations(pod *corev1.Pod, workload *workloadInfo) {
 	pod.Annotations[dtwebhook.AnnotationWorkloadName] = workload.name
 }
 
+func copyMetadataFromNamespace(pod *corev1.Pod, namespace corev1.Namespace, dynakube dynatracev1beta2.DynaKube) {
+	copyMetadataAccordingToCustomRules(pod, namespace, dynakube)
+	copyMetadataAccordingToPrefix(pod, namespace)
+}
+
+func copyMetadataAccordingToPrefix(pod *corev1.Pod, namespace corev1.Namespace) {
+	for key, value := range namespace.Annotations {
+		if strings.HasPrefix(key, dynatracev1beta2.MetadataPrefix) {
+			setPodAnnotationIfNotExists(pod, key, value)
+		}
+	}
+}
+
+func copyMetadataAccordingToCustomRules(pod *corev1.Pod, namespace corev1.Namespace, dynakube dynatracev1beta2.DynaKube) {
+	for _, rule := range dynakube.Status.MetadataEnrichment.Rules {
+		var valueFromNamespace string
+
+		var exists bool
```

Comment:
do you need this bool or is it enough and clearer to check if `valueFromNamespace` is not empty?

Reply:
I would keep it yes, otherwise you would swallow explicitly set empty values